### PR TITLE
feat(testing): set default User-Agent string for simulated requests

### PR DESCRIPTION
**Pull request by kgriffs**
# Summary of Changes

Revert the change to not set a default User-Agent string on the WSGI side, and implement it for ASGI as well.

### CODE REVIEW

File: tests/test_request_attrs.py [hunk 1]
```diff
@@ -690,13 +690,13 @@ def test_date_missing(self, asgi, attr):
...
-        ('If-Range', 'Wed, 21 Oct 2015 07:28:00 GMT', 'if_range', None),
-        ('User-Agent', 'testing/3.0', 'user_agent', None),
-        ('Referer', 'https://www.google.com/', 'referer', None),
+        # ('Accept', 'x-falcon', 'accept', '*/*'),
+        # ('Authorization', 'HMAC_SHA1 c590afa9bb59191ffab30f223791e82d3fd3e3af', 'auth', None),
+        # ('Content-Type', 'text/plain', 'content_type', None),
+        # ('Expect', '100-continue', 'expect', None),
+        # ('If-Range', 'Wed, 21 Oct 2015 07:28:00 GMT', 'if_range', None),
+        ('User-Agent', 'testing/3.0', 'user_agent', 'falcon-client/' + falcon.__version__),
+        # ('Referer', 'https://www.google.com/', 'referer', None),
```
Comment (CaselIT) [inline tests/test_request_attrs.py:699]: The comments here seem a debug leftover

**Review by CaselIT — APPROVED** [review]
Ok to me . There is just a debug leftover in a test

File: tests/asgi/test_testing_asgi.py [hunk 2]
```diff
@@ -115,6 +115,44 @@ def test_cookies_jar():
...
+
+    scope = testing.create_scope(headers={'user-agent': ua})
+    assert dict(scope['headers'])[b'user-agent'] == ua.encode()
+
+    req = testing.create_asgi_req(headers={'user-agent': ua})
+    assert req.user_agent == ua
+
+
+def test_create_scope_default_ua_modify_global():
+    default_ua = 'URL/Emacs Emacs/26.3 (x86_64-pc-linux-gnu)'
```
Comment (vytas7) [inline tests/asgi/test_testing_asgi.py:139]: :rofl:

Comment (vytas7) [reply tests/asgi/test_testing_asgi.py:139]: https://www.gnu.org/software/emacs/manual/html_mono/eww.html

File: tests/test_request_attrs.py (hunk 1 above)
Comment (vytas7) [reply tests/test_request_attrs.py]: I've uncommented those fixtures. Let's see what happens :boom:

**Review by vytas7 — APPROVED** [review]
:+1:

**Review by CaselIT — APPROVED** [review]

**Review by vytas7 — APPROVED** [review]

**Pull request merged**